executor/executor.go

```diff
@@ -109,39 +109,42 @@ func Run(path string, env []string, emitter screwdriver.Emitter, build screwdriv
 	f.Write([]byte(shargs))
 
 	cmds := build.Commands
+	var exitErr error
 
 	for _, cmd := range cmds {
-		if err := api.UpdateStepStart(buildID, cmd.Name); err != nil {
-			return fmt.Errorf("Updating step start %q: %v", cmd.Name, err)
-		}
+		if exitErr == nil || cmd.AlwaysRun == "true" {
+			if err := api.UpdateStepStart(buildID, cmd.Name); err != nil {
+				exitErr = fmt.Errorf("Updating step start %q: %v", cmd.Name, err)
+			}
 
-		// Create step script file
-		stepFilePath := "/tmp/step.sh"
-		if err := createShFile(stepFilePath, cmd); err != nil {
-			return fmt.Errorf("Writing to step script file: %v", err)
-		}
+			// Create step script file
+			stepFilePath := "/tmp/step.sh"
+			if err := createShFile(stepFilePath, cmd); err != nil {
+				exitErr = fmt.Errorf("Writing to step script file: %v", err)
```

Comment:
I've added `continue` statements in each place where it would otherwise be returning. This will ensure the `alwaysRun` logic works. Does this make sense?